### PR TITLE
Use fsdp module to initialize precision scalar for fsdp native

### PR DESCRIPTION
**Pull request by rohitgr7**
## What does this PR do?

Fairscale shouldn't be a requirement when using fsdp_native, so we should rely on fsdp package from torch itself.

## Before submitting

- [ ] Was this **discussed/approved** via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/Lightning-AI/lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- [x] Did you make sure to **update the documentation** with your changes? (if necessary)
- [ ] Did you write any **new necessary tests**? (not for typos and docs)
- [x] Did you verify new and **existing tests pass** locally with your changes?
- [ ] Did you list all the **breaking changes** introduced by this pull request?
- [x] Did you **update the CHANGELOG**? (not for typos, docs, test updates, or minor internal changes/refactors)

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing, make sure you have read the [review guidelines](https://github.com/Lightning-AI/lightning/wiki/Review-guidelines). In short, see the following bullet-list:

- [x] Is this pull request ready for review? (if not, please submit in draft mode)
- [x] Check that all items from **Before submitting** are resolved
- [x] Make sure the title is self-explanatory and the description concisely explains the PR
- [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?

Make sure you had fun coding 🙃

cc @borda @carmocca @justusschock @awaelchli @akihironitta @rohitgr7

### CODE REVIEW

File: src/pytorch_lightning/CHANGELOG.md [hunk 1]
```diff
@@ -70,6 +70,9 @@ The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/).
 - Fixed dtype inference during gradient norm computation ([#14051](https://github.com/Lightning-AI/lightning/pull/14051))
 
 
+- Fixed to use fsdp module to initialize precision scalar for fsdp native ([#14092](https://github.com/Lightning-AI/lightning/pull/14092))
```
Comment (carmocca) [inline src/pytorch_lightning/CHANGELOG.md:73]:
```suggestion
- Avoid requiring the fairscale package to use precision with the fsdp native strategy ([#14092](https://github.com/Lightning-AI/lightning/pull/14092))
```

File: src/pytorch_lightning/CHANGELOG.md [hunk 2]
```diff
@@ -8,7 +8,7 @@ The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/).
 
 ### Added
 
--
+- Added ``FullyShardedNativeNativeMixedPrecisionPlugin`` to handle precision for ``DDPFullyShardedNativeStrategy`` ([#14092](https://github.com/Lightning-AI/lightning/pull/14092))
```
Comment (carmocca) [inline src/pytorch_lightning/CHANGELOG.md:11]:
```suggestion
- Added `FullyShardedNativeNativeMixedPrecisionPlugin` to handle precision for `DDPFullyShardedNativeStrategy` ([#14092](https://github.com/Lightning-AI/lightning/pull/14092))
```

**Review by carmocca — APPROVED** [review]
The fix looks good.

The inheritance hierarchy calls to be refactored here, but the suggested patch seems like the path of least resistance for a bug-fix.
I wonder if the introduced plugin should be marked as protected for that reason.

Also, an orthogonal question is, should we deprecate fairscale's version soon? Or is it still going to be used as a cutting-edge testing ground? Do you have any suggestion in this regard @SeanNaren?

**Comment by rohitgr7** [comment]
> I wonder if the introduced plugin should be marked as protected for that reason.

I don't think so since users should have access to all kinds of plugins we use within strategies for customization.

**Review by kaushikb11 — APPROVED** [review]

**Review by Borda — APPROVED** [review]

**Review by awaelchli — APPROVED** [review]
Retrospectively, we should have named the fsdp strategy FairscaleFSDP or something like that. Normally, everything else we have always implies that it is "native" anyway.

File: src/pytorch_lightning/CHANGELOG.md [hunk 3]
```diff
@@ -95,6 +98,9 @@ The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/).
 - Fixed saving hyperparameters in a composition where the parent class is not a `LightningModule` or `LightningDataModule` ([#14151](https://github.com/Lightning-AI/lightning/pull/14151))
 
 
+- Avoided requiring the fairscale package to use precision with the fsdp native strategy ([#14092](https://github.com/Lightning-AI/lightning/pull/14092))
```
Comment (Felonious-Spellfire) [inline src/pytorch_lightning/CHANGELOG.md:101]:
```suggestion
- Avoided requiring the FairScale package to use precision with the fsdp native strategy ([#14092](https://github.com/Lightning-AI/lightning/pull/14092))
```

**Pull request merged**